dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/YieldingContext.kt

```diff
@@ -0,0 +1,39 @@
+package it.unibo.collektive.aggregate.api
+
+/**
+ * Context for yielding operations.
+ * Yielding operations means that an operator can send a [Initial] value to the neighbours,
+ * but return a [Return] value to the caller.
+ */
+class YieldingContext<Initial, Return> {
+
+    /**
+     * Express the lambda [toReturn] when the computation is done.
+     * It can be used with checks after the invocation.
+     * ## Example
+     * ```
+     * val result = sharing(0) {
+     *   val maxValue = it.maxBy { v -> v.value }.value
+     *   maxValue.yielding { "A string" }
+     * }
+     * result // result: Kotlin.String
+     * ```
+     * ```
+     * val result = sharing(0) {
+     *   val max = it.maxBy { v -> v.value }.value
+     *   max.yielding { "Hello".takeIf { min > 1 } }
+     * }
+     * result // result: Kotlin.String?
+     * ```
+     * The call of [yielding] as the last statement of the body of a yielding operator,
+     * sent to the neighbours the [Initial] value (from the extension receiver),
+     * but returns the [toReturn] value.
```

Comment:
```suggestion
     * Computes [toReturn] after the data exchange operation is complete.
     * ## Example
     * ```
     * sharing(0) { // Sent to neighbors: kotlin.Int
     *     it.max(Int.MIN_VALUE).yielding { it.toString() }
     * }
     * result // result: kotlin.String
     * ```
     * ```
     * val result = sharing(0) {
     *     val max = it.max(Int.MIN_VALUE)
     *     max.yielding { max.toString().takeIf { max > 1 } }
     * }
     * result // result: kotlin.String?
     * ```
     * Calling [yielding] in the body effectively performs the information exchange with neighbors,
     * preparing the local [Initial] value (from the extension receiver) to be sent away,
     * but returns the value produced by [toReturn].
```